pool/noop.go

```diff
@@ -0,0 +1,50 @@
+package pool
+
+import (
+	"context"
+
+	"github.com/mongodb/amboy"
+	"github.com/pkg/errors"
+)
+
+type noopPool struct {
+	isStarted bool
+	queue     amboy.Queue
+}
+
+// NewNoop creates a runner implementation that has no workers, but
+// satisfies the workers and semantics of the Runner interface to
+// support queues deployments that have insert only queues.
+func NewNoop() amboy.Runner { return new(noopPool) }
+
+func (p *noopPool) Started() bool { return p.isStarted }
+
+func (p *noopPool) Start(ctx context.Context) error {
+	if err := ctx.Err(); err != nil {
+		return errors.WithStack(err)
+	}
+
+	if p.queue == nil {
+		return errors.New("cannot start pool without set queue")
+	}
+
+	p.isStarted = true
+	return nil
+}
+
+func (p *noopPool) SetQueue(q amboy.Queue) error {
+	if q == nil {
```

Comment:
This is mutexed in other implementations.

Reply:
I think you only need the mutex in other implementations because the queue is used from the worker threads, but here it never would be, so it's not an issue and won't trip the race detector (which is why we added the mutex, I think the race is really theoretical.) 

I'm inclined to avoid the mutex unless the race detector fails.